ocamltest/ocamltest_stdlib.ml

```diff
@@ -84,22 +84,36 @@ end
 module Sys = struct
   include Sys
 
-  let run_system_command prog args =
-    let command = Filename.quote_command prog args in
-    match Sys.command command with
-    | 0 -> ()
-    | _ as exitcode ->
-      Printf.eprintf "System command %s failed with status %d\n%!"
-        command exitcode;
-      exit 3
-
-  let mkdir dir =
-    if not (Sys.file_exists dir) then
-      run_system_command "mkdir" [dir]
+  let erase_file path =
+    try Sys.remove path
+    with Sys_error _ when Sys.win32 && Ocamltest_config.libunix <> None ->
+      (* Deal with read-only attribute on Windows. Ignore any error from chmod
+         so that the message always come from Sys.remove *)
+      let () = try Unix.chmod path 0o666 with Sys_error _ -> () in
+      Sys.remove path
+
+  let rm_rf path =
+    let rec erase path =
+      if Sys.is_directory path
+      then Array.iter (fun entry -> erase (Filename.concat path entry))
+                      (Sys.readdir path)
+      else erase_file path
+    in
+      try if Sys.file_exists path then erase path
+      with Sys_error _ ->
+        raise (Sys_error (Printf.sprintf "rm_rf failed on %S" path))
 
   let rec make_directory dir =
     if Sys.file_exists dir then ()
-    else (make_directory (Filename.dirname dir); mkdir dir)
+    else let () = make_directory (Filename.dirname dir) in
+         if not (Sys.file_exists dir) then
+           Sys.mkdir dir 0o777
+         else ()
+
+  let make_directory dir =
+    try make_directory dir
+    with Sys_error _ ->
+      raise (Sys_error (Printf.sprintf "make_directory failed on %S" dir))
```

Comment:
Same comment as for `rm_rf`.  Letting the original `Sys_error` exception float up would be better in my opinion.